app/Http/Controllers/SignUpController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use DB;
+use App\Http\Requests;
+use Auth;
+use Otp\GoogleAuthenticator;
+
+class SignUpController extends Controller
+{
+    public function signUpPage()
+    {
+        return view('signup');
+    }
+
+    public function signUp(Request $request)
+    {
+        $email = strtolower($request->input('email'));
+        $password1 = $request->input('password1');
+        $password2 = $request->input('password2');
+        if ($password1 != $password2) {
+            return view('signup', ['errorMessage' => 'Passwords do not match.']);
+        } else {
+            $users = DB::table('users')->where('email', $email)->get();
```

Comment:
Is there any reason to use a direct table query instead of the app\User model? The returned result will be an array of rows, not actual users that can access eloquent properties.